### ISSUE TITLE
error compiling oiio 1.7.x on centos 6

### ISSUE THREAD

**Issue by remusvrm** [OpenImageIO/oiio#1970]
### Problem

I am trying to compile oiio version `1.7.19` on centos 6, gcc `4.4.7`, but I am getting an error:
```
[ 24%] Building CXX object src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/openexr.imageio/exrinput.cpp.o
cd /local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/build/src/libOpenImageIO && /usr/bin/c++   -DEMBED_PLUGINS=1 -DOpenImageIO_EXPORTS -DUSE_BOOST_ASIO=1 -DUSE_FREETYPE -DUSE_GIF -DUSE_JPEG_TURBO=1 -DUSE_OCIO=1 -DUSE_OPENEXR_VERSION2=1 -DUSE_OPENJPEG -DUSE_WEBP=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -O3 -DNDEBUG -fPIC -I/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/build/include/OpenImageIO -I/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/include -I/remote/install/openEXR/2.2.0_static/centos-6_x86-64/include -I/remote/install/openEXR/2.2.0_static/centos-6_x86-64/include/OpenEXR -isystem /remote/install/boost/1.54.0_shared/centos-6_x86-64/include -I/remote/install/opencolorIO/1.0.9/centos-6_x86-64/include -I/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/build/src/libOpenImageIO -I/usr/include/openjpeg -I/usr/include/freetype2 -I/usr/include/freetype2/freetype    -Wall -Werror -fno-math-errno -UUSE_FIELD3D -o CMakeFiles/OpenImageIO.dir/__/openexr.imageio/exrinput.cpp.o -c /local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp
In file included from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/x86_64-redhat-linux/bits/c++allocator.h:34,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/allocator.h:48,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/string:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/locale_classes.h:42,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/ios_base.h:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ios:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/istream:40,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/fstream:40,
                 from /local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:35:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h: In instantiation of ‘__gnu_cxx::new_allocator<const std::basic_string<char, std::char_traits<char>, std::allocator<char> > >’:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/allocator.h:87:   instantiated from ‘std::allocator<const std::basic_string<char, std::char_traits<char>, std::allocator<char> > >’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:71:   instantiated from ‘std::_Vector_base<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<const std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:171:   instantiated from ‘std::vector<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<const std::basic_string<char, std::char_traits<char>, std::allocator<char> > > >’
/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:1206:   instantiated from here
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:79: error: ‘const _Tp* __gnu_cxx::new_allocator<_Tp>::address(const _Tp&) const [with _Tp = const std::basic_string<char, std::char_traits<char>, std::allocator<char> >]’ cannot be overloaded
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:76: error: with ‘_Tp* __gnu_cxx::new_allocator<_Tp>::address(_Tp&) const [with _Tp = const std::basic_string<char, std::char_traits<char>, std::allocator<char> >]’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h: In instantiation of ‘__gnu_cxx::new_allocator<const OpenImageIO::v1_7::TypeDesc>’:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/allocator.h:87:   instantiated from ‘std::allocator<const OpenImageIO::v1_7::TypeDesc>’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:71:   instantiated from ‘std::_Vector_base<const OpenImageIO::v1_7::TypeDesc, std::allocator<const OpenImageIO::v1_7::TypeDesc> >’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:171:   instantiated from ‘std::vector<const OpenImageIO::v1_7::TypeDesc, std::allocator<const OpenImageIO::v1_7::TypeDesc> >’
/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:1206:   instantiated from here
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:79: error: ‘const _Tp* __gnu_cxx::new_allocator<_Tp>::address(const _Tp&) const [with _Tp = const OpenImageIO::v1_7::TypeDesc]’ cannot be overloaded
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:76: error: with ‘_Tp* __gnu_cxx::new_allocator<_Tp>::address(_Tp&) const [with _Tp = const OpenImageIO::v1_7::TypeDesc]’
/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp: In member function ‘virtual bool OpenImageIO::v1_7::OpenEXRInput::read_native_deep_scanlines(int, int, int, int, int, OpenImageIO::v1_7::DeepData&)’:
/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:1206: error:   initializing argument 3 of ‘void OpenImageIO::v1_7::DeepData::init(int, int, OpenImageIO::v1_7::array_view<const OpenImageIO::v1_7::TypeDesc, 1ul>, OpenImageIO::v1_7::array_view<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, 1ul>)’
In file included from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/x86_64-redhat-linux/bits/c++allocator.h:34,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/allocator.h:48,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/string:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/locale_classes.h:42,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/ios_base.h:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ios:43,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/istream:40,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/fstream:40,
                 from /local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:35:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h: At global scope:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h: In instantiation of ‘__gnu_cxx::new_allocator<const unsigned int>’:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/allocator.h:87:   instantiated from ‘std::allocator<const unsigned int>’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:71:   instantiated from ‘std::_Vector_base<const unsigned int, std::allocator<const unsigned int> >’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/stl_vector.h:171:   instantiated from ‘std::vector<const unsigned int, std::allocator<const unsigned int> >’
/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/src/openexr.imageio/exrinput.cpp:1233:   instantiated from here
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:79: error: ‘const _Tp* __gnu_cxx::new_allocator<_Tp>::address(const _Tp&) const [with _Tp = const unsigned int]’ cannot be overloaded
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/ext/new_allocator.h:76: error: with ‘_Tp* __gnu_cxx::new_allocator<_Tp>::address(_Tp&) const [with _Tp = const unsigned int]’
make[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/openexr.imageio/exrinput.cpp.o] Error 1
make[2]: Leaving directory `/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/build'
make[1]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
make[1]: Leaving directory `/local/software/download/openimageIO/1.7.19/src/oiio-Release-1.7.19/build'
make: *** [all] Error 2
```

I am using the following variables to configure it:
```
cmake -fPIC \
      -DVERBOSE=1 \
      -DOIIO_BUILD_CPP11=0 \
      -DOIIO_BUILD_CPP14=0 \
      -DUSE_CPP11=0 \
      -DBOOST_ROOT=${BOOST_ROOT} \
      -DBOOST_INCLUDEDIR=${BOOST_INCLUDEDIR} \
      -DOPENEXR_HOME=${OPENEXR_HOME} \
      -DILMBASE_HOME=${ILMBASE_HOME} \
      -DUSE_PYTHON=1 \
      -DPYTHON_VERSION=${PYTHON_VERSION} \
      -DPYTHON_LIBRARY=${PYTHON_LIBRARY} \
      -DPYTHON_INCLUDE_DIR=${PYTHON_INCLUDE_DIR} \
      -DUSE_FFMPEG=1 \
      -DUSE_OPENJPEG=1 \
      -DUSE_GIF=1 \
      -DUSE_OCIO=1 \
      -DUSE_OPENCV=0 \
      -DOCIO_PATH=${OPENCOLORIO_PATH} \
      -DOCIO_INCLUDES=${OCIO_INCLUDES} \
      -DUSE_NUKE=1 \
      -DNuke_ROOT=${NUKE_HOME} \
      -DNUKE_VERSION=${NUKE_VERSION} \
      -DUSE_LIBRAW=1 \
      -DLIBRAW_PATH=${LIBRAW_PATH} \
      -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_DIR} \
      ..
```

### Versions

* OIIO `1.7.19` 
* OS: centos `6.8`
* C++ compiler: `4.4.7 20120313`
* python: `2.7.11`
* boost: `1.54.0` shared libraries
* cmake: `3.3.2`
* openexr: `2.2.0`
* OCIO: `1.0.9`
* LibRaw: `0.18.12`

**Comment by hobbes1069** [comment]
If I remember correctly Larry mentioned you'd need a C++ standard flag which I don't see in your output... Perhaps adding "-std=c++03" would be enough?

**Comment by remusvrm** [comment]
@hobbes1069: Adding `-std=c++03` will cause other problems:
```
  cc1plus: error: unrecognized command line option "-std=c++03"

  gmake[1]: *** [CMakeFiles/cmTC_996ef.dir/testCXXCompiler.cxx.o] Error 1
```

**Comment by lgritz** [comment]
I don't know the answer. Those error messages don't look familiar to me and I don't have any hypothesis on what's going wrong.

Is there any way for you to use a newer compiler? Like gcc 4.8? (Even in C++03 mode.)

I'm worried about gcc 4.4. The VFX Platform recommendations (http://www.vfxplatform.com/) have dictated gcc 4.8.2 as far back as 2015, so I think it's fair to say that OIIO 1.7 (first released in 2016) has had minimal use with gcc 4.4 and may well rely on things that weren't supported or were buggy in versions prior to 4.8.

**Comment by remusvrm** [comment]
All right. If I build against a new gcc version it works.
Seems to be a compiler issue.
